src/dynd/array.cpp

```diff
@@ -224,22 +228,77 @@ void nd::array::flag_as_immutable()
 nd::array nd::array::p(const char *name) const
 {
   if (!is_null()) {
-    ndt::type dt = get_type();
-    std::map<std::string, nd::callable> properties = dt.get_array_properties();
-
-    callable p = properties[name];
-    if (!p.is_null()) {
-      return p(eval()); // ToDo: Replace eval() here with an implicit cast
+    const ndt::type &dt = get_dtype();
+
+    switch (dt.get_id()) {
+    case complex_float32_id:
+    case complex_float64_id:
+      return callable_registry[name](eval());
+    case struct_id:
+      return nd::make_field_access_kernel(dt, name)(eval());
```

Comment:
This is of course inefficient. We could memoize this later in callable_registry.cpp when we update the kernel.